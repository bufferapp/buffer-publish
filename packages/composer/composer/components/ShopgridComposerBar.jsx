import React from 'react';
import PropTypes from 'prop-types';
import styles from './css/ShopgridComposerBar.css';

import ComposerActionCreators from '../action-creators/ComposerActionCreators';

class ShopgridComposerBar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      shopgridLink: props.shopgridLink || '',
    };

    this.onChange = this.onChange.bind(this);
    this.shouldShowShopgridBar = this.shouldShowShopgridBar.bind(this);
  }

  onChange(event) {
    this.setState({ shopgridLink: event.target.value || ''});
    ComposerActionCreators.updateDraftShopgridLink(this.props.draftId, event.target.value);
  }

  shouldShowShopgridBar() {
    const {
      isInstagram,
      hasShopgridFlip,
      selectedInstagramProfiles,
      isBusinessUser,
    } = this.props;

    const hasInstagramSelected = selectedInstagramProfiles && selectedInstagramProfiles.length >= 1;

    return hasInstagramSelected && isInstagram && hasShopgridFlip && isBusinessUser;
  }

  render () {
    return (
      this.shouldShowShopgridBar() &&
      <div className={styles.shopgridComposerBar}>
        <div className={styles.shopgridFieldContainer}>
          <span className={styles.shopgridFieldLabel}>Link </span>
          <span className={styles.shopgridAutocompleteContainer}>
            <input
              className={styles.shopgridFieldInput}
              onChange={this.onChange}
              onBlur={this.onChange}
              name="shopgridLink"
              placeholder="Website or Product URL"
              value={this.state.shopgridLink}
            />
          </span>
        </div>
      </div>
    );
  }
}

ShopgridComposerBar.propTypes = {
  draftId: PropTypes.string.isRequired,
  shopgridLink: PropTypes.string,
  hasShopgridFlip: PropTypes.bool.isRequired,
  isInstagram: PropTypes.bool.isRequired,
  selectedInstagramProfiles: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  isBusinessUser: PropTypes.bool.isRequired,
};

ShopgridComposerBar.defaultProps = {
  shopgridLink: null,
};


export default ShopgridComposerBar;
