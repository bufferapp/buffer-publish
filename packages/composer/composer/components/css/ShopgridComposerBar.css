.shopgridComposerBar {
    padding: 10px;
    width: 100%;
    box-sizing: border-box;
    border-bottom-left-radius: 3px;
    border-bottom-right-radius: 3px;
    float: left;
}

.shopgridFieldContainer {
    float: left;
    padding-top: 1px;
    position: relative;
    display: flex;
    width: 100%;
}

.shopgridFieldLabel {
    font-weight: 600;
    font-size: 12px;
    color: #343e47;
    min-width: 80px;
    padding-top: 6px;
    position: relative;
}

.shopgridAutocompleteContainer {
    flex-grow: 1;
}

.shopgridFieldInput {
    border-radius: 4px;
    padding-left: 8px;
    padding-top: 8px;
    padding-bottom: 8px;
    padding-right: 50px;
    margin-left: 4px;
    border: 1px solid #E6EBEF;
    box-sizing: border-box;
    font-weight: 500;
    font-size: 12px;
    width: 100%;
}

.questionIcon {
    display: inline;
    float: right;
    margin-top: 2px;
    margin-right: 3px;
    cursor: pointer;
}
