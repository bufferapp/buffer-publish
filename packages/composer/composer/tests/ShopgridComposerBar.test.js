/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import renderer from 'react-test-renderer';
import ShopgridComposerBar from '../components/ShopgridComposerBar';

jest.mock('../action-creators/ComposerActionCreators');
jest.mock('../utils/LocationFinder');
jest.mock('lodash.debounce', () => jest.fn(fn => fn));

const showShopgridBar = {
  isInstagram: true,
  hasShopgridFlip: true,
  selectedInstagramProfiles: [{},{}],
  isBusinessUser: true,
};

describe('Whole component', () => {
  it('renders default state', () => {
    const tree = renderer.create(
      <ShopgridComposerBar
        {...showShopgridBar}

        draftId={'123'}
        shopgridLink={undefined}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with shopgrid link when set', () => {
    const tree = renderer.create(
      <ShopgridComposerBar
        {...showShopgridBar}
        draftId={'123'}
        shopgridLink={'test'}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('doesn\'t render when shopgrid flip is not set', () => {
    const tree = renderer.create(
      <ShopgridComposerBar
        {...showShopgridBar}
        hasShopgridFlip={false}
        draftId={'123'}
        shopgridLink={undefined}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('doesn\'t render for non business users', () => {
    const tree = renderer.create(
      <ShopgridComposerBar
        {...showShopgridBar}
        isBusinessUser={false}
        draftId={'123'}
        shopgridLink={undefined}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

