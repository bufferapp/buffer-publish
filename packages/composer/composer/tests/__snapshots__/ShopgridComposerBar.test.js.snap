// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Whole component doesn't render for non business users 1`] = `null`;

exports[`Whole component doesn't render when shopgrid flip is not set 1`] = `null`;

exports[`Whole component renders default state 1`] = `
<div
  className="shopgridComposerBar"
>
  <div
    className="shopgridFieldContainer"
  >
    <span
      className="shopgridFieldLabel"
    >
      Link 
    </span>
    <span
      className="shopgridAutocompleteContainer"
    >
      <input
        className="shopgridFieldInput"
        name="shopgridLink"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Website or Product URL"
        value=""
      />
    </span>
  </div>
</div>
`;

exports[`Whole component renders with shopgrid link when set 1`] = `
<div
  className="shopgridComposerBar"
>
  <div
    className="shopgridFieldContainer"
  >
    <span
      className="shopgridFieldLabel"
    >
      Link 
    </span>
    <span
      className="shopgridAutocompleteContainer"
    >
      <input
        className="shopgridFieldInput"
        name="shopgridLink"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Website or Product URL"
        value="test"
      />
    </span>
  </div>
</div>
`;
